checks/burrowsc/burrow.go

```diff
@@ -0,0 +1,236 @@
+				{
+					Description: name,
+					Result:      healthchecks.CRITICAL,
+					Details:     fmt.Sprintf("Invalid response. Code: %d, Body: %s", resp.StatusCode, responseBody),
+				},
+			},
+		}
+	}
+
+	lagResponse := LagResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&lagResponse)
+	if err != nil {
+		return healthchecks.StatusList{
+			StatusList: []healthchecks.Status{
+				{
+					Description: name,
+					Result:      healthchecks.CRITICAL,
+					Details:     fmt.Sprintf("Error decoding json response: %s", err.Error()),
+				},
+			},
+		}
+	}
+
+	lagStatus := lagResponse.LagStatus
+	var s *healthchecks.Status
+
+	// If topic is nil, check the status of the entire consumer group instead of a specific partition
+	if b.Topic == nil {
+		s = checkGroupStatus(name, lagStatus, b.CriticalLagThreshold)
+	} else {
+		for _, partition := range lagStatus.Partitions {
+			if partition.Topic == *b.Topic {
+				s = checkPartitionStatus(name, lagStatus, partition, b.CriticalLagThreshold)
+				break
+			}
+		}
+
+		// Topic was not found in consumer group partitions
+		if s == nil {
+			s = &healthchecks.Status{
+				Description: name,
+				Result:      healthchecks.WARNING,
+				Details:     fmt.Sprintf("Topic %s not found in group %s on cluster %s", *b.Topic, b.ConsumerGroup, b.Cluster),
+			}
+		}
+	}
+
+	return healthchecks.StatusList{
+		StatusList: []healthchecks.Status{
+			*s,
+		},
+	}
+}
+
+func checkGroupStatus(name string, lagStatus LagStatus, criticalLagThreshold *int64) *healthchecks.Status {
+	// If critical lag threshold is specified and exceeded, return a critical alert
+	if criticalLagThreshold != nil && lagStatus.TotalLag > *criticalLagThreshold {
+		return &healthchecks.Status{
+			Description: name,
+			Result:      healthchecks.CRITICAL,
+			Details:     fmt.Sprintf("%s exceeds threshold", formatConsumerGroupDetails(lagStatus)),
+		}
+	}
+
+	// If critical lag threshold was not exceeded or not specified, check Burrow consumer group status
+	return &healthchecks.Status{
+		Description: name,
+		Result:      getAlertLevel(lagStatus.Status),
+		Details:     formatConsumerGroupDetails(lagStatus),
+	}
+}
+
+func checkPartitionStatus(name string, lagStatus LagStatus, partition Partition, criticalLagThreshold *int64) *healthchecks.Status {
+	// If critical lag threshold is specified and exceeded, return a critical alert
+	if criticalLagThreshold != nil && partition.CurrentLag > *criticalLagThreshold {
+		return &healthchecks.Status{
+			Description: name,
+			Result:      healthchecks.CRITICAL,
+			Details:     fmt.Sprintf("%s exceeds threshold", formatPartitionDetails(partition, lagStatus)),
+		}
+	}
+
+	// If critical lag threshold was not exceeded or not specified, check Burrow partition status
+	return &healthchecks.Status{
+		Description: name,
+		Result:      getAlertLevel(partition.Status),
+		Details:     formatPartitionDetails(partition, lagStatus),
+	}
+}
+
+func getAlertLevel(status string) healthchecks.AlertLevel {
```

Comment:
Can we important this from somewhere else?

Reply:
No, don't think we can import these from anywhere. This is converting the Burrow statuses to the health check statuses.

Reply:
ah sorry i thought it was just going "OK" to the alert level OK